Assets/MixedRealityToolkit/InputModule/Scripts/Utilities/Interactions/TwoHandManipulatable.cs

```diff
@@ -0,0 +1,631 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+// #defines for features that are not yet implemented
+#define TODO_ROTATE_FACE_USER
+
+using MixedRealityToolkit.Common;
+using MixedRealityToolkit.Common.Extensions;
+using MixedRealityToolkit.InputModule.EventData;
+using MixedRealityToolkit.InputModule.Focus;
+using MixedRealityToolkit.InputModule.InputHandlers;
+using MixedRealityToolkit.InputModule.InputSources;
+using MixedRealityToolkit.UX.BoundingBoxes;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using UnityEngine.Assertions;
+
+
+namespace MixedRealityToolkit.InputModule.Utilities.Interations
+{
+    /// <summary>
+    /// This script allows for an object to be movable, scalable, and rotatable with one or two hands. 
+    /// You may also configure the script on only enable certain manipulations. The script works with 
+    /// both HoloLens' gesture input and immersive headset's motion controller input.
+    /// See Assets/MixedRealityToolkit-Examples/Input/Readme/README_TwoHandManipulationTest.md
+    /// for instructions on how to use the script.
+    /// </summary>
+    public class TwoHandManipulatable : MonoBehaviour, IInputHandler, ISourceStateHandler
+    {
+        // Event that gets raised when user begins manipulating the object
+        public event Action StartedManipulating;
+        // Event that gets raised when the user ends manipulation
+        public event Action StoppedManipulating;
+        // Bounding Box visual on manipulation interaction
+        public GameObject GlobalBoundingBox;
```

Comment:
We should probably encapsulate this and serialize the private field

Reply:
Resolved.